internal/system/agent/services.go

```diff
@@ -131,6 +136,49 @@ func getMetrics(services []string, ctx context.Context) (MetricsRespMap, error)
 	return m, nil
 }
 
+func getHealth(services []string, ctx context.Context) HealthRespMap {
+
+	h := HealthRespMap{}
+	h.Health = map[string]interface{}{}
+
+	config := registry.Config{
+		Host:            Configuration.Registry.Host,
+		Port:            Configuration.Registry.Port,
+		Type:            Configuration.Registry.Type,
+		ServiceKey:      internal.SystemManagementAgentServiceKey,
+		ServiceHost:     Configuration.Service.Host,
+		ServicePort:     Configuration.Service.Port,
+		ServiceProtocol: Configuration.Service.Protocol,
+		CheckInterval:   Configuration.Service.CheckInterval,
+		CheckRoute:      clientsMod.ApiPingRoute,
+		Stem:            internal.ConfigRegistryStem,
+	}
+	client, err := factory.NewRegistryClient(config)
+	if err != nil {
+		logs.LoggingClient.Error("could not create registry client", "errMsg", err.Error())
+		return h
+	}
+
+	for _, service := range services {
+
+		h.Health[service] = ""
+
+		if !isKnownServiceKey(service) {
+			logs.LoggingClient.Warn("unknown service found getting health", "service name", service)
+		}
+
+		err := client.IsServiceAvailable(service)
```

Comment:
- With the addition of `go-mod-registry` (to the go.mod file) in place and `go-mod-registry` downloaded into the module cache: I would be interested in hearing about any testing steps you may have carried out (using Postman, etc.) to trace the flow that involves the `Registry` service.

Reply:
I started Consul, populated it with configuration from `config-seed`, then started `core-data` with Consul configuration, then started the SMA and tested health. I confirmed that health is green when `core-data` was running, and red when I killed `core-data`